beacon_node/rest_api/src/beacon.rs

```diff
@@ -128,13 +183,29 @@ pub struct StateResponse<T: EthSpec> {
 /// Will not return a state if the request slot is in the future. Will return states higher than
 /// the current head by skipping slots.
 pub fn get_state<T: BeaconChainTypes + 'static>(req: Request<Body>) -> ApiResult {
-    let beacon_chain = req
-        .extensions()
-        .get::<Arc<BeaconChain<T>>>()
-        .ok_or_else(|| ApiError::ServerError("Beacon chain extension missing".to_string()))?;
-
-    let query_params = ["root", "slot"];
-    let (key, value) = UrlQuery::from_request(&req)?.first_of(&query_params)?;
+    let beacon_chain = get_beacon_chain_from_request::<T>(&req)?;
+    let head_state = get_head_state(beacon_chain.clone())?;
```

Comment:
We don't need to build the caches here.

The caches are not returned in the serialized object (they're transparent to users), so there's no need to have them built.